kratos/tensor_adaptors/equation_ids_tensor_adaptor.cpp

```diff
@@ -0,0 +1,119 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/data_type_traits.h"
+#include "utilities/parallel_utilities.h"
+#include "tensor_adaptor_utils.h"
+
+// Include base h
+#include "equation_ids_tensor_adaptor.h"
+
+namespace Kratos {
+
+namespace EquationIdsTensorAdaptorHelpers {
+
+template <class TContainerType>
+void CollectData(
+    Kratos::span<int> Span,
+    const unsigned int Stride,
+    const TContainerType& rContainer,
+    const ProcessInfo& rProcessInfo)
+{
+    IndexPartition<IndexType>(rContainer.size()).for_each(std::vector<IndexType>{}, [Span, &rContainer, &rProcessInfo, Stride](const auto Index, auto& rTLS){
+        const auto& r_entity = *(rContainer.begin() + Index);
+        r_entity.EquationIdVector(rTLS, rProcessInfo);
+
+        KRATOS_DEBUG_ERROR_IF_NOT(rTLS.size() == Stride)
+            << "Non-uniform equation id vectors are found in container having " << rContainer.size()
+            << " " << ModelPart::Container<TContainerType>::GetEntityName() << "(s)"
+            << " [ Required size of equation ids = " << Stride << ", found EquationIds = "
+            << rTLS << " ].\n";
+
+        std::copy(rTLS.begin(), rTLS.end(), Span.data() + Index * Stride);
```

Comment:
it is not .data().begin()?

Reply:
Both can be used. since std::copy works on the `iterator::operator*`, so both will work. This is more readable I guess.